src/main/java/dev/sigstore/fulcio/client/FulcioClient.java

```diff
@@ -86,43 +89,57 @@ public FulcioClient build() {
    *
    * @param cr certificate request parameters
    * @return a {@link SigningCertificate} from fulcio
-   * @throws IOException if the http request fials
-   * @throws CertificateException if returned certificates could not be decoded
    */
   public SigningCertificate SigningCert(CertificateRequest cr)
-      throws IOException, CertificateException {
-    URI fulcioEndpoint = serverUrl.resolve(SIGNING_CERT_PATH);
-
-    HttpRequest req =
-        HttpClients.newHttpTransport(httpParams)
-            .createRequestFactory()
-            .buildPostRequest(
-                new GenericUrl(fulcioEndpoint),
-                ByteArrayContent.fromString("application/json", cr.toJsonPayload()));
-
-    req.getHeaders().setAccept("application/pem-certificate-chain");
-    req.getHeaders().setAuthorization("Bearer " + cr.getIdToken());
-
-    HttpResponse resp = req.execute();
-    if (resp.getStatusCode() != 201) {
-      throw new IOException(
-          String.format(
-              "bad response from fulcio @ '%s' : %s", fulcioEndpoint, resp.parseAsString()));
-    }
-
-    String sctHeader = resp.getHeaders().getFirstHeaderStringValue("SCT");
-    try (InputStream content = resp.getContent()) {
-      var signingCert =
-          SigningCertificate.newSigningCertificate(
-              CharStreams.toString(new InputStreamReader(content, resp.getContentCharset())),
-              sctHeader);
-      if (signingCert.getDetachedSct().isEmpty() && !signingCert.hasEmbeddedSct() && requireSct) {
-        throw new IOException(
-            "no signed certificate timestamps were found in response from Fulcio");
+      throws InterruptedException, CertificateException, IOException {
+    // TODO: we should reuse this message channel across various executions of this client and
```

Comment:
what's the impact of not doing this? should there be a bug?

Reply:
yeah I think we would need an autoclosable or something so users could do:
```
try (var connection = FulcioClient.newConnection()) {
  connection.signingCert()
  ...
  connection.differentSigningCert()
}
```
Right now it's a bit expensive, but I don't think dangerous.

Reply:
I don't actually know a reason why someone would want to keep the connection open for too long with fulcio. We probably only need one cert to sign a bunch of artifacts.

Reply:
updated the comment